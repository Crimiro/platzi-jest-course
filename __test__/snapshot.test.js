import { getCharacter } from '../snapshot';
import rick from '../rick.json';

describe('Es hora de las intantaneas', () => {
  test('Snapshot', () => {
    expect(getCharacter(rick)).toMatchSnapshot();
  });
  test('Siempre fallará la instantanea', () => {
    const user = {
      createdAt: new Date(),
      id: Math.floor(Math.random() * 20)
    }
    expect(user).toMatchSnapshot();
  });

  test('Tenemos una excepción', () => {
    const user = {
      id: Math.floor(Math.random() * 20),
      name: 'Crix Rosales'
    }
    expect(user).toMatchSnapshot({
      id: expect.any(Number)
    });
  });
});