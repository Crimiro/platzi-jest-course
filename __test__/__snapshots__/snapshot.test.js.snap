// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Es hora de las intantaneas Siempre fallará la instantanea 1`] = `
Object {
  "createdAt": 2019-06-17T00:47:54.936Z,
  "id": 11,
}
`;

exports[`Es hora de las intantaneas Snapshot 1`] = `
Array [
  Object {
    "gender": "Male",
    "id": 1,
    "name": "Rick Sanchez",
    "species": "Human",
    "status": "Alive",
  },
]
`;

exports[`Es hora de las intantaneas Tenemos una excepción 1`] = `
Object {
  "id": Any<Number>,
  "name": "Crix Rosales",
}
`;
